### ISSUE TITLE
Error message is confusing when missing semicolon in data or parameters block

### ISSUE THREAD

**Issue by jgabry** [stan-dev/stanc3#677]
Unlike in other blocks, if the user forgets a semicolon in the data block or parameters block the error message doesn't mention a semicolon at all. For example, the following Stan program is missing a semicolon in the parameters block but gives the confusing error message below:

```stan
parameters {
 real theta
}
model {
 theta ~ std_normal();
}
```

```
   -------------------------------------------------
     1:  
     2:  parameters {
     3:   real theta
                    ^
     4:  }
     5:  model {
   -------------------------------------------------

Only top-level variable declarations are allowed in data and parameter blocks.
```

I think it would be hard for any new user to figure out that they're missing a semicolon based on an error message about "only top-level variable declarations are allowed". Is there a way to mention the missing semicolon in the message?

**Comment by rybern** [comment]
This is a quick fix, see #678

**Comment by jgabry** [comment]
Awesome, thanks @rybern!
